src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_params.py

```diff
@@ -190,6 +190,7 @@ def get_vm_size_completion_list(prefix, action, parsed_args, **kwargs):  # pylin
     register_cli_argument(scope, 'admin_username', help='Username for the VM.', arg_group='Authentication')
     register_cli_argument(scope, 'admin_password', help="Password for the VM if authentication type is 'Password'.", arg_group='Authentication')
     register_cli_argument(scope, 'ssh_key_value', help='SSH public key or public key file path.', completer=FilesCompleter(), type=file_type, arg_group='Authentication')
+    register_cli_argument(scope, 'custom_data', help='Custom init script file or text (cloud-init, cloud-config, etc..)', completer=FilesCompleter(), type=file_type)
```

Comment:
In that case I'm fine with custom-data. 'user-data' is just as vague but breaks with Xplat so might as well stay the same.